Sources/Diagnostics/DiagnosticsFileHandler.swift

```diff
@@ -0,0 +1,142 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  DiagnosticsFileHandler.swift
+//
+//  Created by Nacho Soto on 6/16/23.
+
+import Foundation
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+protocol DiagnosticsFileHandlerType: AnyObject {
+
+    func getEntries() async -> [DiagnosticsEvent]
+
+    func appendEvent(diagnosticsEvent: DiagnosticsEvent) async
+
+    func cleanSentDiagnostics(diagnosticsSentCount: Int) async
+
+    func emptyDiagnosticsFile() async
+
+}
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+actor DiagnosticsFileHandler: DiagnosticsFileHandlerType {
+
+    private let fileHandler: FileHandler
+
+    init?() {
+        do {
+            self.fileHandler = try .init(Self.diagnosticsFile)
+        } catch {
+            Logger.error("Initialization error: \(error.localizedDescription)")
+            return nil
+        }
+    }
+
+    init(_ fileHandler: FileHandler) {
+        self.fileHandler = fileHandler
+    }
+
+    func appendEvent(diagnosticsEvent: DiagnosticsEvent) async {
+        guard let jsonString = try? diagnosticsEvent.encodedJSON else {
+            Logger.error("Failed to serialize diagnostics event to JSON")
+            return
+        }
+
+        await fileHandler.append(line: jsonString)
+    }
+
+    func getEntries() async -> [DiagnosticsEvent] {
+        var entries: [DiagnosticsEvent] = []
+
+        if #available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *) {
+            do {
+                for try await line in try await fileHandler.readLines() {
+                    if let event = decodeDiagnosticsEvent(from: line) {
+                        entries.append(event)
+                    }
+                }
+            } catch {
+                Logger.error("Failed to read lines from file: \(error.localizedDescription)")
+            }
+        } else {
```

Comment:
I wonder if we should just not support diagnostics in iOS < 15... I think it should be fine?

Reply:
yeah, I'd be fine with that if there's any extra cost to supporting iOS < 15. If we already have the code I wouldn't remove it, but if we need to add anything else, we might as well cut it

Reply:
ok, I cut the support to 15, it will simplify the code, and it is probably overengineering